app/tests/emails_tests/test_tasks.py

```diff
@@ -168,16 +169,136 @@ def test_filter_recipients(subscription_type, expected_recipients):
     u3.user_profile.receive_notification_emails = False
     u3.user_profile.save()
 
-    filtered_userset = filter_recipients(
+    send_standard_email_batch(
         recipients=[inactive_user, u1, u2, u3],
         subscription_type=subscription_type,
+        markdown_message="foo",
+        site=Site.objects.get_current(),
+        subject="bar",
     )
-    assert inactive_user not in filtered_userset
+
+    sent_emails = [email.to[0] for email in mail.outbox]
+
     expected_users = [
-        value
-        for boolean, value in zip(
+        user.email
+        for boolean, user in zip(
             expected_recipients, [u1, u2, u3], strict=True
         )
         if boolean
     ]
-    assert expected_users == filtered_userset
+    assert expected_users == sent_emails
+
+
+@pytest.mark.parametrize(
+    "subscription_type",
+    [
+        EmailSubscriptionTypes.NEWSLETTER,
+        EmailSubscriptionTypes.NOTIFICATION,
+        EmailSubscriptionTypes.SYSTEM,
+    ],
+)
+@pytest.mark.django_db
+def test_cannot_email_inactive_users(subscription_type):
+    inactive_user = UserFactory(is_active=False)
+
+    with pytest.raises(ValueError) as error:
+        create_email_object(
+            recipient=inactive_user,
+            connection=get_connection(),
+            site=Site.objects.get_current(),
+            markdown_message="foo",
+            subject="bar",
+            subscription_type=subscription_type,
+        )
+
+    assert str(error.value) == "Inactive users cannot be emailed"
+
+
+@pytest.mark.django_db
+def test_can_always_email_system_messages():
+    user = UserFactory(is_active=True)
+
+    user.user_profile.receive_newsletter = True
+    user.user_profile.receive_notification_emails = True

```

Comment:
Shouldn't these be set to `False` to test that we can always send system emails?

Reply:
Good catch!